org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/service/WildcardServiceTest.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+package org.osgi.test.junit5.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.osgi.framework.BundleContext;
+import org.osgi.test.common.annotation.InjectBundleContext;
+import org.osgi.test.common.annotation.InjectService;
+import org.osgi.test.common.annotation.InjectService.AnyService;
+import org.osgi.test.common.dictionary.Dictionaries;
+import org.osgi.test.common.service.ServiceAware;
+import org.osgi.test.junit5.context.BundleContextExtension;
+import org.osgi.test.junit5.types.Foo;
+
+@ExtendWith(BundleContextExtension.class)
+@ExtendWith(ServiceExtension.class)
+public class WildcardServiceTest {
+
+	@BeforeAll
+	public static void beforeAll(@InjectBundleContext BundleContext bundleContext) throws Exception {
+		bundleContext.registerService(B.class, new B() {}, Dictionaries.dictionaryOf("a", "b"));
+		bundleContext.registerService(Object.class, new Foo() {}, Dictionaries.dictionaryOf("a", "b"));
+
+	}
+
+	@InjectService(filter = "(a=b)")
+	ServiceAware<?>			field_ServiceAwareWildcard_Default;
+
+	@InjectService(filter = "(a=b)", service = AnyService.class)
+	ServiceAware<?>			field_ServiceAwareWildcard_AnyService;
+
+	@InjectService(filter = "(a=b)")
+	ServiceAware<Object>	field_ServiceAwareObject_Default;
+
+	@InjectService(filter = "(a=b)", service = AnyService.class)
+	ServiceAware<Object>	field_ServiceAwareObject_AnyService;
+
+	@Test
+	public void test_field_ServiceAwareWildcard_Default() throws Exception {
+		assertThat(field_ServiceAwareWildcard_Default.getServices()).hasSize(1)
+			.hasOnlyElementsOfTypes(Foo.class);
+	}
+
+	@Test
+	public void test_field_ServiceAwareWildcard_AnyService() throws Exception {
+		assertThat(field_ServiceAwareWildcard_AnyService.getServices()).hasSize(2)
+			.hasOnlyElementsOfTypes(B.class, Foo.class);
+	}
+
+	@Test
+	public void test_field_ServiceAwareObject_Default() throws Exception {
+		assertThat(field_ServiceAwareObject_Default.getServices()).hasSize(1)
+			.hasOnlyElementsOfTypes(Foo.class);
+	}
+
+	@Test
+	public void test_field_ServiceAwareObject_AnyService() throws Exception {
+		assertThat(field_ServiceAwareObject_AnyService.getServices()).hasSize(2)
+			.hasOnlyElementsOfTypes(B.class, Foo.class);
+	}
+
+	@Test
+	public void test_param_ServiceAwareWildcard_Default(
+		@InjectService(filter = "(a=b)") ServiceAware<?> param_ServiceAwareWildcard_Default) throws Exception {
+		assertThat(param_ServiceAwareWildcard_Default.getServices()).hasSize(1)
+			.hasOnlyElementsOfTypes(Foo.class);
+	}
+
+	public void test_param_ServiceAwareWildcard_AnyService(
+		@InjectService(filter = "(a=b)", service = AnyService.class) ServiceAware<?> param_ServiceAwareWildcard_AnyService)
+		throws Exception {
+		assertThat(field_ServiceAwareWildcard_AnyService.getServices()).hasSize(2)
```

Comment:
You need to assert on `param_ServiceAwareWildcard_AnyService`. Not the field.

Reply:
oh , thank you. done